code/modules/shuttle/shuttle_projector.dm

```diff
@@ -0,0 +1,172 @@
+/// Projects a shuttle with visual juice while it docks/launches with vis_contents
+/obj/shuttle_projector
+	layer = LOWER_SHUTTLE_MOVEMENT_LAYER
+	plane = LOWER_SHUTTLE_MOVEMENT_PLANE
+	appearance_flags = KEEP_TOGETHER
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	icon = 'icons/effects/alphacolors.dmi'
+	icon_state = "transparent"
+
+	/// The mobile port we're projecting
+	var/obj/docking_port/mobile/shuttle_port
+	/// The bottom left turf of the bounding box where the shuttle will dock in the stationary port
+	var/turf/bottom_left
+
+/obj/shuttle_projector/Initialize(mapload, obj/docking_port/mobile/shuttle_port, obj/docking_port/stationary/stationary_port, inbound, total_animate_time = null)
+	. = ..()
+
+	if (!istype(shuttle_port))
+		stack_trace("Invalid shuttle_port for shuttle_projector!")
+		return INITIALIZE_HINT_QDEL
+	if (!istype(stationary_port))
+		stack_trace("Invalid stationary_port for shuttle_projector!")
+		return INITIALIZE_HINT_QDEL
+	src.shuttle_port = shuttle_port
+
+	// Get the mobile ports turfs to project
+	var/list/all_transit_turfs = shuttle_port.return_ordered_turfs(shuttle_port.x, shuttle_port.y, shuttle_port.z, shuttle_port.dir)
+
+	var/list/initial_shuttle_turfs = list()
+	for (var/T in all_transit_turfs)
+		var/turf/transit_turf = T
+		// We don't want to project any empty space turfs
+		if (!istype(transit_turf, /turf/open/space))
+			initial_shuttle_turfs += transit_turf
```

Comment:
This lost ones on catwalk

Reply:
Now projects everything in the shuttle bounds and hide transit turfs